src/main/java/de/tum/in/www1/artemis/config/websocket/WebsocketConfiguration.java

```diff
@@ -228,4 +249,10 @@ private boolean isParticipationOwnedByUser(Principal principal, Long participati
         StudentParticipation participation = participationService.findOneStudentParticipation(participationId);
         return participation.isOwnedBy(principal.getName());
     }
+
+    private boolean isUserTAOrHigherForExercise(Principal principal, Long exerciseId) {
+        Exercise exercise = exerciseService.findOne(exerciseId);
+        User user = userService.getUserWithGroupsAndAuthorities(principal.getName());
+        return authorizationCheckService.isAtLeastInstructorForExercise(exercise, user);
```

Comment:
The method name suggests that it is checked whether the user is a TA or above but the actual check is done for instructor or above.

Reply:
Very good catch! I'll fix it.